### PR TITLE
dtls: cleanup some code

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
- factorize BIO_CTRL_DGRAM_SET_NEXT_TIMEOUT calls.
- simplify a return type
- style nits

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by FdaSilvaYY** [comment]
Spinoff from #14344

**Review by t8m — APPROVED** [review]

File: ssl/d1_lib.c [hunk 1]
```diff
@@ -225,7 +225,7 @@ long dtls1_ctrl(SSL *ssl, int cmd, long larg, void *parg)
 
     switch (cmd) {
     case DTLS_CTRL_GET_TIMEOUT:
-        if (dtls1_get_timeout(s, &t) != NULL) {
+        if (dtls1_get_timeout(s, &t) != 0) {
```
Comment (tmshort) [inline ssl/d1_lib.c:228]: This is now a boolean? So just:
```
if (dtls1_get_timeout(s, &t)) {

File: ssl/d1_lib.c [hunk 2]
```diff
@@ -308,22 +314,20 @@ OSSL_TIME *dtls1_get_timeout(SSL_CONNECTION *s, OSSL_TIME *timeleft)
...
+    return 1;
 }
 
 int dtls1_is_timer_expired(SSL_CONNECTION *s)
 {
     OSSL_TIME timeleft;
 
     /* Get time left until timeout, return false if no timer running */
-    if (dtls1_get_timeout(s, &timeleft) == NULL) {
+    if (dtls1_get_timeout(s, &timeleft) == 0)
```
Comment (tmshort) [inline ssl/d1_lib.c:325]: Boolean?
```
if (!dtlsq_get_timeout(s, &timeleft))

File: ssl/d1_lib.c [hunk 3]
```diff
@@ -338,14 +342,12 @@ static void dtls1_double_timeout(SSL_CONNECTION *s)
...
-    struct timeval tv;
-
     /* Reset everything */
     s->d1->timeout_num_alerts = 0;
     s->d1->next_timeout = ossl_time_zero();
     s->d1->timeout_duration_us = 1000000;
-    tv = ossl_time_to_timeval(s->d1->next_timeout);
-    BIO_ctrl(s->rbio, BIO_CTRL_DGRAM_SET_NEXT_TIMEOUT, 0, &tv);
+    /* set s->d1->next_timeout into bio interface */
+    dtls1_bio_set_next_timeout(s->rbio, s->d1);
```
Comment (tmshort) [inline ssl/d1_lib.c:350]: The comment here (and a similar one elsewhere) probably not necessary as the function name is quite descriptive.

**Comment by FdaSilvaYY** [comment]
@tmshort : fixup pushed, thanks for the review.

**Review by tmshort — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master branch. Thank you for your contribution.

**Pull request closed without merging**